posts/2019/11/Variable-Explorer.md

```diff
@@ -0,0 +1,73 @@
+
+This made very cumbersome to use this functionality, which was the reason why arbitrary Python objects were hidden by default from the Variable Explorer view.  
+In this version we decided to integrate the excellent [objbrowser](https://github.com/titusjan/objbrowser) project, by Pepijn Kenter ([@titusjan](https://github.com/titusjan)), which provides a tree-like view of Python objects, for a much simpler and comfortable way to inspect them.
+
+![Python Viewer Metadata](/images/spyder-variable-explorer/python-viewer-metadata.png)
+
+As can be seen above, this viewer will also allow users to have access to some extra metadata about the inspected object, such as its documentation, source code and the file that holds it.  
+It is very important to note that this work was accomplished thanks to the generosity of Pepijn, who kindly changed the license of [objbrowser](https://github.com/titusjan/objbrowser) to allow us integrating it with Spyder.
+To expose this new functionality we also decided to turn off the Variable Explorer option called “Exclude unsupported types” and introduce a new one called “Exclude callables and modules” (turned on by default) to display a much larger fraction of objects after a file or cell execution.
+
+![Exclude callables modules](/images/spyder-variable-explorer/exclude-callables-modules.png)
+
+Finally, we added a context-menu action to allow using the object explorer viewer for all objects, regardless if they have a default viewer or not.
+
+![View object explorer](/images/spyder-variable-explorer/view-object-explorer.png)
+
+## Multi-index support in the dataframe viewer
+
+One of the first enhancements we did to the Variable Explorer in Spyder 4 was adding support for multi-dimensional indexes to its dataframe viewer. In Spyder 3 the support was very limited, which made inspecting those indexes a painful experience:
+
+![Multi-index support](/images/spyder-variable-explorer/multi-index-support.png)
+
+For Spyder 4 we took advantage of the work done by Scott Hansen ([@firecat53](https://github.com/firecat53)) and Yuri D'Elia ([@wavexx](https://github.com/wavexx)) in their [gtabview](https://github.com/TabViewer/gtabview) project for this very purpose. The main element we took from it was its improved management of column and table headings, which allowed us to display the index shown above in a much nicer way:
+
+![Table headings](/images/spyder-variable-explorer/table-headings.png)
+
+## Fuzzy filtering of variables
+
+This version also includes the ability to filter variables in the current namespace to focus only on the ones you are interested in. This filtering is done in a fuzzy way by matching any letter of text entered in the search field with the type and name of all available variables.
+To access this functionality you need to press the keyboard shortcut `Ctrl+F` when the Variable Explorer has focus or by pressing the magnifying glass icon on top,
```

Comment:
```suggestion
To access this functionality, click the search icon in the Variable Explorer toobar, or press `Ctrl+F` (`Cmd-F` on macOS) when the Variable Explorer has focus.
```

- Simplify and clarify description and make it more user-friendly and idiomatic
- Add Mac keyboard shortcut

Reply:
I think you meant toolbar!

Reply:
Yes, thanks! Fixed now!